modules/mesh-layers/src/simple-mesh-layer/simple-mesh-layer.js

```diff
@@ -162,36 +161,30 @@ export default class MeshLayer extends Layer {
   }
 
   updateState({props, oldProps, changeFlags}) {
-    const attributeManager = this.getAttributeManager();
-
-    // super.updateState({props, oldProps, changeFlags});
-    if (changeFlags.dataChanged) {
-      attributeManager.invalidateAll();
+    if (props.mesh !== oldProps.mesh || props.fp64 !== oldProps.fp64) {
+      if (this.state.model) {
+        this.state.model.delete();
+      }
+      if (props.mesh) {
+        this.setState({model: this.getModel(props.mesh)});
+      }
+      this.getAttributeManager().invalidateAll();
     }
 
-    this._updateFP64(props, oldProps);
-
     if (props.texture !== oldProps.texture) {
       this.setTexture(props.texture);
     }
-  }
-
-  _updateFP64(props, oldProps) {
-    if (props.fp64 !== oldProps.fp64) {
-      if (this.state.model) {
-        this.state.model.delete();
-      }
-
-      this.setState({model: this.getModel(this.context.gl)});
-
-      this.setTexture(this.state.texture);
 
-      const attributeManager = this.getAttributeManager();
-      attributeManager.invalidateAll();
+    if (changeFlags.dataChanged) {
+      super.updateState({props, oldProps, changeFlags});
```

Comment:
You can drop the if: https://github.com/uber/deck.gl/blob/master/modules/layers/src/icon-layer/icon-layer.js#L124